internal/pkg/events/incoming/ceilometer.go

```diff
@@ -98,6 +105,45 @@ func (evt *CeilometerEvent) ParseEvent(data string) error {
 		log.Fatal(err)
 		return err
 	}
+	// transforms traits key into map[string]interface{}
```

Comment:
Storage to ES currently uses the sanitized json string, so I do not think these changes are even being used: https://github.com/infrawatch/smart-gateway/blob/80caab3837a3876381947cdfc8a141b6543b57f7/internal/pkg/events/events.go#L321

The ElasticClient Create() function will also need to be modified from using BodyString() to BodyJson(): https://github.com/infrawatch/smart-gateway/blob/80caab3837a3876381947cdfc8a141b6543b57f7/internal/pkg/saelastic/client.go#L149

Reply:
I took the liberty of making those changes, and it passes!

```
=== RUN   TestStoreCeilometerEvent
=== RUN   TestStoreCeilometerEvent/Store_event_with_differing_trait_types
newPayload: map[event_type:image.delete generated:2020-03-06T14:13:29.497096 message_id:084c0bca-0d19-40c0-a724-9916e4815845 message_signature:77e798b842991f9c0c35bda265fdf86075b4a1e58309db1d2adbf89386a3859e raw:map[] traits:map[created_at:2020-03-06T14:01:07 deleted_at:2020-03-06T14:13:29 name:cirros project_id:0f500647077b47f08a8ca9181e9b7aef resource_id:c4f7e00b-df85-4b77-9e1a-26a1de4d5735 service:image.localhost size:1.3287936e+07 status:deleted user_id:0f500647077b47f08a8ca9181e9b7aef]]
--- PASS: TestStoreCeilometerEvent (0.03s)
    --- PASS: TestStoreCeilometerEvent/Store_event_with_differing_trait_types (0.03s)
```